docs/src/configuration.md

```diff
@@ -52,26 +48,51 @@ with:
 $ mpiexecjl --project=/path/to/project -n 20 julia script.jl
 ```
 
-## Using a system-provided MPI
+## Using MPIPreferences.jl
+
+MPI.jl uses [Preferences.jl](https://github.com/JuliaPackaging/Preferences.jl) to
+allow the user to choose which MPI implementation to use for a project. This provides
+a single source of truth that can be used for `JLL` (Julia packages providing C libraries)
+that link against MPI, as well as localizing the choice of MPI implementation to a project.
+
+Users can use the provided [`use_system_binary`](@ref) or [`use_jll_binary`](@ref)
+to switch MPI implementations.
+
+### Migration from MPI.jl `v0.19`
+
+Prior to MPI.jl `v0.20` environment variables were used to configure which MPI
+library to use. These have now been removed.
 
-### Requirements
+- `JULIA_MPI_BINARY`
+- `JULIA_MPIEXEC`
+- `JULIA_MPI_INCLUDE_PATH`
+- `JULIA_MPI_CFLAGS`
+- `JULIA_MPICC`
+
+### Using a system-provided MPI
+
+#### Requirements
 
 MPI.jl requires a shared library installation of a C MPI library, supporting the MPI 3.0
 standard or later.
 
-### Building
+### Configuration
 
-To use the the system MPI, set the environment variable `JULIA_MPI_BINARY=system` and run
-`Pkg.build("MPI")`. This can be done by:
-```
-julia --project -e 'ENV["JULIA_MPI_BINARY"]="system"; using Pkg; Pkg.build("MPI"; verbose=true)'
+To use the the system MPI, run `MPI.use_system_binary()`.
+This will attempt find and identify any available MPI implementation, and create
+a file called `LocalPreferences.toml` adjacent to the current `Project.toml`.
```

Comment:
You are always using a project :), but yes it would be created in `.julia/environment/v1.7` if you are using the default project.

Reply:
I know, but it took me a while to understand this fact. I just thought it might help users to avoid the "huh, where is the newly created preferences file?" moment. But it is also sort of redundant to the basic Julia docs, so feel free to ignore this suggestion if you do not like it.